test/system/255-auto-update.bats

```diff
@@ -0,0 +1,213 @@
+    run_podman ? auto-update
+    is "$output" ".*invalid auto-update policy.*" "invalid policy setup"
+    run_podman ps -a
+    sleep 2
+    run_podman inspect --format "{{.Image}}" $cname
+    systemctl stop container-$cname
+    rm -f $UNIT_DIR/container-$cname.service
+    is "$output" "$ori_image" "Image hash should not changed."
+}
+
+@test "podman auto-update with multiple services" {
+    fakevalue=$(random_string)
+    run_podman inspect --format "{{.Id}}" $IMAGE
+    img_id="$output"
+    cnames=()
+    cnames_update=()
+    run generate_service $IMAGE quay.io/libpod/alpine:latest top image
+    cname_update="${lines[0]}"
+    cnames+=($cname_update)
+    cnames_update+=($cname_update)
+    echo $output
+
+    run generate_service $IMAGE quay.io/libpod/alpine_nginx:latest top image
+    cname_update_1="${lines[0]}"
+    cnames+=($cname_update_1)
+    cnames_update+=($cname_update_1)
+    echo $output
+
+    run generate_service $IMAGE quay.io/libpod/alpine:latest top
+    cname_default="${lines[0]}"
+    cnames+=($cname_default)
+    echo $output
+
+    run generate_service $IMAGE quay.io/libpod/alpine:latest top disabled
+    cname_disabled="${lines[0]}"
+    cnames+=($cname_disabled)
+    echo $output
+
+    run generate_service $IMAGE quay.io/libpod/alpine:latest top ""
+    cname_empty="${lines[0]}"
+    cnames+=($cname_empty)
+    echo $output
+
+    run generate_service $IMAGE quay.io/libpod/alpine:latest top $fakevalue
+    cname_fakevalue="${lines[0]}"
+    cnames+=($cname_fakevalue)
+    echo $output
+
+    sleep 2
+    run_podman ? auto-update
+    sleep 2
+    count=0
+    while read line; do
+       if [[ "$line" =~ "Trying to pull" ]]; then
+           echo "Count shoud be added"
+           ((count+=1))
+       fi
+    done <<< "$output"
+    is "$output" ".*invalid auto-update policy.*" "invalid policy setup"
+    is "$output" ".*1 error occurred.*" "invalid policy setup"
+    is "$output" ".*$cname_update_1.*" "container with auto-update policy image updated"
+    is "$output" ".*$cname_update.*" "container with auto-update policy image updated"
+    is "$count" "2" "There are two images being updated."
+
+
+    for cname in "${cnames[@]}"; do
+        run_podman inspect --format "{{.Image}}" $cname
+        if [[ "${cnames_update[@]}" =~ "$cname" ]]; then
+            [[ "$output" != "$img_id" ]]
+        else
+            is "$output" "$img_id" "Image is not changed."
+        fi
+        systemctl stop container-$cname
+        rm -f $$UNIT_DIR/container-$cname.service
+    done
+}
+
+@test "podman auto-update using systemd" {
+    run generate_service $IMAGE quay.io/libpod/alpine:latest top image
+    cname=${lines[0]}
+    echo $output
+    run_podman inspect --format "{{.Image}}" $cname
+    ori_image=$output
+
+    sleep 2
+    cp $UNIT_DIR/podman-auto-update.timer $UNIT_DIR/podman-auto-update.timer-bk
+    sed -i 's/OnCalendar=.*/OnCalendar=*-*-* *:*:0\/2/g' $UNIT_DIR/podman-auto-update.timer
```

Comment:
I really have to encourage you not to do this. It's one thing to add scratch system files; quite another to muck with existing distro-installed and -needed files. I strongly urge you to find a different way to do this, such as creating a `podman-test-randomname.timer` in `/etc/systemd` instead.

Reply:
OK. Will set up an timer for testing. Thanks for your advice.